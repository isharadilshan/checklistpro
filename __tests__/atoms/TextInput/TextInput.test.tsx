import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import TextInput from '../../../src/components/atoms/TextInput';

configure({adapter: new Adapter()});

const props = {
  label: 'TEST TEXT INPUT',
  value: 'test@gmail.com',
  onChangeText: jest.fn(),
  error: true,
  errorText: 'THIS IS ERROR TEXT',
  secureTextEntry: true,
};

describe('when rendering', () => {
  const wrapper = shallow(<TextInput {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
