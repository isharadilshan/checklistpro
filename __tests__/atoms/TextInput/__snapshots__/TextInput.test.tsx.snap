// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<View
  style={
    {
      "marginVertical": 12,
      "width": "100%",
    }
  }
>
  <withTheme(undefined)
    error={true}
    label="TEST TEXT INPUT"
    onChangeText={[MockFunction]}
    secureTextEntry={true}
    style={
      {
        "backgroundColor": "#191c1f",
      }
    }
    theme={
      {
        "colors": {
          "placeholder": "#007d7d",
          "primary": "#0F4C81",
          "text": "#ffffff",
        },
      }
    }
    value="test@gmail.com"
  />
  <Text
    style={
      [
        {
          "fontSize": 14,
          "paddingHorizontal": 4,
          "paddingTop": 4,
        },
        {
          "color": "#e76372",
        },
      ]
    }
  >
    THIS IS ERROR TEXT
  </Text>
</View>
`;
